lib/vehicle.rb

```diff
@@ -5,14 +5,20 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registration_date,
+              :plate_type
+              
+  attr_accessor :registration_date, :plate_type
```

Comment:
It's redundant to specify both `attr_reader` and `attr_accessor` for the same attributes (`registration_date` and `plate_type`). If you want them to be both readable and writable, you can use `attr_accessor` alone.